runtime/sunvmi/sunvmi.c

```diff
@@ -612,13 +612,30 @@ getClassContextIterator(J9VMThread * currentThread, J9StackWalkState * walkState
 JNIEXPORT void JNICALL
 JVM_GC_Impl(void)
 {
-	J9VMThread *currentThread = VM.javaVM->internalVMFunctions->currentVMThread(VM.javaVM);
+	J9InternalVMFunctions *vmFuncs = VM.javaVM->internalVMFunctions;
+	J9VMThread *currentThread = vmFuncs->currentVMThread(VM.javaVM);
+	J9MemoryManagerFunctions *mmFuncs = VM.javaVM->memoryManagerFunctions;
 
 	Trc_SunVMI_GC_Entry(currentThread);
 
-	VM.javaVM->internalVMFunctions->internalEnterVMFromJNI(currentThread);
-	VM.javaVM->memoryManagerFunctions->j9gc_modron_global_collect(currentThread);
-	VM.javaVM->internalVMFunctions->internalExitVMToJNI(currentThread);
+#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+#else /* J9VM_INTERP_ATOMIC_FREE_JNI */
+	vmFuncs->internalAcquireVMAccess(currentThread);
+#endif /* J9VM_INTERP_ATOMIC_FREE_JNI */
+
+	mmFuncs->j9gc_modron_global_collect(currentThread);
+	mmFuncs->j9gc_modron_global_collect(currentThread);
```

Comment:
Perhaps a comment is warranted here explaining why `j9gc_modron_global_collect()` is run twice?